Autopilot/Src/altimeter.cpp

```diff
@@ -0,0 +1,153 @@
+#include "altimeter.hpp"
+
+//Address for the MS5637. Needs to be shifted for write/read (ref:datasheet)
+#define MS5637_ADDR 0x76
+#define MS5637_WRITE_ADDR 0xEC
+#define MS5637_READ_ADDR 0xED
+#define MS5637_I2C I2C2 //Port... I think?
+//registers
+//Hex codes for polling, ref: Datasheet pg.9
+#define MS5637_RESET_CMD 0x1E
+#define MS5637_START_PRESSURE_CONVERSION 0x40
+#define MS5637_START_TEMP_CONVERSION 0x50
+#define MS5637_READ_ADC 0x00
+#define MS5637_INTERN_MEM_ADDRESS 0x00 //There isn't actually one. Datasheet and stack exchange confirms
+
+//PROM commands
+#define MS5637_PROM_C1 0xA2
+#define MS5637_PROM_C2 0xA4
+#define MS5637_PROM_C3 0xA6
+#define MS5637_PROM_C4 0xA8
+#define MS5637_PROM_C5 0xAA
+#define MS5637_PROM_C6 0xAC
+
+bool MS5637::isI2CBusDefined = false;
+
+static I2C_HandleTypeDef* hi2c;
```

Comment:
I'd have to include i2c.h in the altimeter.hpp file then for it to compile without errors.

Reply:
t's fine to include things in the header so long as you have a reason to. This looks like a reason. We're just trying to prevent including files that serve no purpose.

But I don't really want to get hung up on that atm. I'll take a look at the rest. If things work, we can just move on.